lib/Utils/PlatformMac.cpp

```diff
@@ -0,0 +1,214 @@
+  if (!fs::is_directory(SDKs))
+    return false;
+
+
+  // Try to get the SDK for whatever version of OS X is currently running
+  // Seems to make more sense to get the currently running SDK so headers
+  // and any loaded libraries will match.
+
+  int32_t majorVers = -1, minorVers = -1;
+#ifndef CLING_SWVERS_PARSE_ONLY
+ #pragma clang diagnostic push
+ #pragma clang diagnostic ignored "-Wdeprecated-declarations"
+  if (void *core = ::dlopen(
+          "/System/Library/Frameworks/CoreServices.framework/CoreServices",
+          RTLD_LAZY)) {
+    typedef ::OSErr (*GestaltProc)(::OSType, ::SInt32 *);
+    if (GestaltProc Gestalt = (GestaltProc)::dlsym(core, "Gestalt")) {
+      if (Gestalt(gestaltSystemVersionMajor, &majorVers) == ::noErr) {
+        if (Gestalt(gestaltSystemVersionMinor, &minorVers) != ::noErr)
+          minorVers = -1;
+      } else
+        majorVers = -1;
+    }
+    ::dlclose(core);
+  }
+ #pragma clang diagnostic pop
+#endif
+
+  if (majorVers == -1 || minorVers == -1) {
+    const std::string SWVers = ReadSingleLine("sw_vers | grep ProductVersion"
+                                              " | awk '{print $2}'");
+    if (!SWVers.empty()) {
+      if (::sscanf(SWVers.c_str(), "%d.%d", &majorVers, &minorVers) != 2) {
+        majorVers = -1;
+        minorVers = -1;
+      }
+    }
+  }
+
+  if (majorVers != -1 && minorVers != -1) {
+    if (getISysRootVersion(SDKs, majorVers, minorVers, sysRoot,
+            Verbose ? "match the version of OS X running"
+                    : nullptr)) {
+      return true;
+    }
+  }
+
+
+#define GET_ISYSROOT_VER(maj, min) \
+  if (getISysRootVersion(SDKs, maj, min, sysRoot, Verbose ? \
+                 "match what cling was compiled with" : nullptr)) \
+    return true;
+
+  // Try to get the SDK for whatever cling was compiled with
+  #if defined(MAC_OS_X_VERSION_10_11)
+    GET_ISYSROOT_VER(10, 11);
+  #elif defined(MAC_OS_X_VERSION_10_10)
+    GET_ISYSROOT_VER(10, 10);
+  #elif defined(MAC_OS_X_VERSION_10_9)
+    GET_ISYSROOT_VER(10, 9);
+  #elif defined(MAC_OS_X_VERSION_10_8)
+    GET_ISYSROOT_VER(10, 8);
+  #elif defined(MAC_OS_X_VERSION_10_7)
+    GET_ISYSROOT_VER(10, 7);
+  #elif defined(MAC_OS_X_VERSION_10_6)
+    GET_ISYSROOT_VER(10, 6);
+  #elif defined(MAC_OS_X_VERSION_10_5)
+    GET_ISYSROOT_VER(10, 5);
+  #elif defined(MAC_OS_X_VERSION_10_4)
+    GET_ISYSROOT_VER(10, 4);
+  #elif defined(MAC_OS_X_VERSION_10_3)
+    GET_ISYSROOT_VER(10, 3);
+  #elif defined(MAC_OS_X_VERSION_10_2)
+    GET_ISYSROOT_VER(10, 2);
+  #elif defined(MAC_OS_X_VERSION_10_1)
+    GET_ISYSROOT_VER(10, 1);
+  #else // MAC_OS_X_VERSION_10_0
+    GET_ISYSROOT_VER(10, 0);
+  #endif
+
+#undef GET_ISYSROOT_VER
+
+  // Nothing left to do but iterate the SDKs directory
+  // copy the paths and then sort for the latest
+  std::error_code ec;
+  std::vector<std::string> srtd;
+  for (fs::directory_iterator it(SDKs, ec), e; it != e; it.increment(ec))
```

Comment:
Yes, https://github.com/root-mirror/cling/pull/53/commits/24540d84ac17b570bd6a4895662df4f0d93884a3 dressed this too.